Adding statefulness when toggling settings

Kyle-Mendes opened the pull request:
Resolves #6 

I put this as a draft because there is some weirdness when toggling between `hidden` and then toggling `paused` that I couldn't figure out. 

But, basically, this will keep the state of the plugin so that when you spawn new pets, they'll respect that state.

Example:
1. PetsSleepingToggle
2. PetsNew a

RESULT: You spawn a sleeping cat

### Issues
- [x] Pausing then spawning a cat results in the cat being 1 row lower than it should be until you unpause
- [x] `PetsNew a` -> `PetsNew b` -> `PetsHiddenToggle` -> `PetsHiddenToggle` -> `PetsPauseToggle` can result in weirdness where one cat is paused and another isn't.

giusgad commented:
I fixed the issue where the pets did not pause after being hidden and unhidden, it was because of the global paused status not being updated when exiting the hidden status.
Also, the reason pets spawn one row lower while paused has something to do with the fact that next_frame is called directly, the workaround I tried (and it worked) was decreasing the row before calling next_frame and increasing it again right after. This did not work with row=1 (a simple if statement would solve it) but I'm thinking there must be a cleaner way and I'll try finding it later today.

P.S. I don't know why my commit is not showing up here, but it is on the branch :)

Kyle-Mendes commented:
Awesome! When I get to my computer I'll take a look. Thanks for finding the issue with setting paused after hidden!

Kyle-Mendes commented:
@giusgad I found the commit :) It was in the main repo while this PR is in my fork.  I pulled in your commit

giusgad commented:
I added another fix for weirdness caused by toggling pause while pets were hidden, but still couldn't figure out why they get displayed one row below normal while paused. If we cannot find a fix for this, in the next couple of days, an easy option would be to not display the pet when created during pause and only show it after the state get unpaused...

Btw, do you maybe know an easier way to send my changes here than having you pull them from my branch?

Kyle-Mendes commented:
> I added another fix for weirdness caused by toggling pause while pets were hidden, but still couldn't figure out why they get displayed one row below normal while paused.
> 
> Edit: fixed the second issue too, we should be ready to merge!

Awesome! I just pulled your changes

> Btw, do you maybe know an easier way to send my changes here than having you pull them from my branch?

Not sure. I think you'd need to commit to my branch in my repo directly, which you're welcome to do!
